include/boost/url/format.hpp

```diff
@@ -0,0 +1,415 @@
+            args.begin(), args.end()));
+}
+
+/** Format arguments into a URL
+
+    Format arguments according to the format
+    URL string into a @ref url_base.
+
+    This overload allows type-erased arguments
+    to be passed as an initializer_list, which
+    is mostly convenient for named parameters.
+
+    All arguments must be convertible to a
+    implementation defined type able to store a
+    type-erased reference to any valid format
+    argument.
+
+    The rules for a format URL string are the same
+    as for a `std::format_string`, where replacement
+    fields are delimited by curly braces.
+
+    The URL components to which replacement fields
+    belong are identified before replacement is
+    applied and any invalid characters for that
+    formatted argument are percent-escaped.
+
+    Hence, the delimiters between URL components,
+    such as `:`, `//`, `?`, and `#`, should be
+    included in the URL format string. Likewise,
+    a format string with a single `"{}"` is
+    interpreted as a path and any replacement
+    characters invalid in this component will be
+    encoded to form a valid URL.
+
+    @par Example
+    @code
+    static_url<30> u;
+    format_to(u, "user/{id}", {{"id", 1}})
+    assert(u.buffer() == "user/1");
+    @endcode
+
+    @par Preconditions
+    All replacement fields must be valid and the
+    resulting URL should be valid after arguments
+    are formatted into the URL.
+
+    Because any invalid characters for a URL
+    component are encoded by this function, only
+    replacements in the scheme and port components
+    might be invalid, as these components do not
+    allow percent-encoding of arbitrary
+    characters.
+
+    @par Exception Safety
+    Strong guarantee.
+
+    @param u An object that derives from @ref url_base.
+    @param fmt The format URL string.
+    @param args Arguments to be formatted.
+
+    @throws system_error
+    `fmt` contains an invalid format string and
+    `u` contains an invalid URL after replacements
+    are applied.
+
+    @par BNF
+    @code
+    replacement_field ::=  "{" [arg_id] [":" (format_spec | chrono_format_spec)] "}"
+    arg_id            ::=  integer | identifier
+    integer           ::=  digit+
+    digit             ::=  "0"..."9"
+    identifier        ::=  id_start id_continue*
+    id_start          ::=  "a"..."z" | "A"..."Z" | "_"
+    id_continue       ::=  id_start | digit
+    @endcode
+
+    @par Specification
+    @li <a href="https://fmt.dev/latest/syntax.html"
+        >Format String Syntax</a>
+
+    @see
+        @ref format.
+
+*/
+inline
+void
+format_to(
+    url_base& u,
+    string_view fmt,
+#ifdef BOOST_URL_DOCS
+    std::initializer_list<__implementation_defined__> args
+#else
+    std::initializer_list<detail::format_arg> args
+#endif
+    )
+{
+    detail::vformat_to(
+        u, fmt, detail::format_args(
+            args.begin(), args.end()));
+}
+
+/** Construct a named argument for a replacement field
```

Comment:
Instead of "construct a named argument" you can say "designate a named argument"